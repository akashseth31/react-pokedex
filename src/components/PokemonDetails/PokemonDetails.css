.overlay {
    position: fixed;
    background: rgba(0, 0, 0, 0.9);
    bottom: 0;
    top: 0;
    left: 0;
    right: 0;
    z-index: 111;
}

.pokemon-details {
    position: absolute;
    width: 100%;
    max-width: 650px;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}

.pokemon-details > button {
    outline: none;
    width: 30px;
    height: 30px;
    position: absolute;
    right: 0;
    top: 0;
    background: #ffffff;
    border-radius: 100%;
    border: 3px solid #FF9441;
    transform: translate(50%, -50%);
    cursor: pointer;
}

.pokemon-details > button::before {
    position: absolute;
    width: 4px;
    height: 18px;
    left: 50%;
    top: 50%;
    background: #FF9441;
    content: '';
    transform: translate(-50%, -50%) rotate(45deg);
}

.pokemon-details > button::after {
    position: absolute;
    width: 4px;
    height: 18px;
    left: 50%;
    top: 50%;
    background: #FF9441;
    content: '';
    transform: translate(-50%, -50%) rotate(-45deg);
}

.pokemon-details-inner {
    border: 3px solid #FF9441;
    border-radius: 10px;
    background-color: #ffffff;
    display: flex;
    justify-content: space-around;
    align-items: center;
}

.pokemon-details-inner aside {
    width: 50%;
    min-height: 300px;
    max-height: 500px;
    overflow-y: auto;
}

.pokemon-details-inner aside.image {
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
}

.pokemon-details-inner aside.details {
    padding: 20px;
    background: #f5f5f5;
    text-transform: uppercase;
    color: #585858;
}

aside.details h3 {
    font-size: 22px;
    color: #FF9441;
}

aside.details h4 {
    font-size: 16px;
}

aside.details section {
    margin-top: 10px;
}

aside.details span {
    display: inline-block;
    font-size: 14px;
    border: 1px solid #FF9441;
    padding: 3px 7px;
    border-radius: 5px;
    margin-right: 5px;
    margin-top: 5px;
}

