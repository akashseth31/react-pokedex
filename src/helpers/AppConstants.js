import error404 from './assets/error-404.png';

export const GET_POKEDEX_DATA = 'http://pokeapi.co/api/v2/pokemon/';
export const LIMIT = 100;
export const NOT_FOUND = error404;

