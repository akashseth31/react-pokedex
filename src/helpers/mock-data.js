export const data = [
  {
    "name": "bulbasaur",
    "url": "https://pokeapi.co/api/v2/pokemon/1/",
    "details": {
      "abilities": [
        {
          "ability": {
            "name": "chlorophyll",
            "url": "https://pokeapi.co/api/v2/ability/34/"
          },
          "is_hidden": true,
          "slot": 3
        },
        {
          "ability": {
            "name": "overgrow",
            "url": "https://pokeapi.co/api/v2/ability/65/"
          },
          "is_hidden": false,
          "slot": 1
        }
      ],
      "height": 7,
      "id": 1,
      "sprites": {
        "back_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png",
        "back_female": null,
        "back_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/1.png",
        "back_shiny_female": null,
        "front_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
        "front_female": null,
        "front_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/1.png",
        "front_shiny_female": null
      },
      "types": [
        {
          "slot": 2,
          "type": {
            "name": "poison",
            "url": "https://pokeapi.co/api/v2/type/4/"
          }
        },
        {
          "slot": 1,
          "type": {
            "name": "grass",
            "url": "https://pokeapi.co/api/v2/type/12/"
          }
        }
      ],
      "weight": 69
    }
  },
  {
    "name": "ivysaur",
    "url": "https://pokeapi.co/api/v2/pokemon/2/",
    "details": {
      "abilities": [
        {
          "ability": {
            "name": "chlorophyll",
            "url": "https://pokeapi.co/api/v2/ability/34/"
          },
          "is_hidden": true,
          "slot": 3
        },
        {
          "ability": {
            "name": "overgrow",
            "url": "https://pokeapi.co/api/v2/ability/65/"
          },
          "is_hidden": false,
          "slot": 1
        }
      ],
      "height": 10,
      "id": 2,
      "sprites": {
        "back_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/2.png",
        "back_female": null,
        "back_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/2.png",
        "back_shiny_female": null,
        "front_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png",
        "front_female": null,
        "front_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/2.png",
        "front_shiny_female": null
      },
      "types": [
        {
          "slot": 2,
          "type": {
            "name": "poison",
            "url": "https://pokeapi.co/api/v2/type/4/"
          }
        },
        {
          "slot": 1,
          "type": {
            "name": "grass",
            "url": "https://pokeapi.co/api/v2/type/12/"
          }
        }
      ],
      "weight": 130
    }
  }
];