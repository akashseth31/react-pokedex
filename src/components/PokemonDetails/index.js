import React from 'react';
import PropTypes from 'prop-types';
import './PokemonDetails.css';
import { NOT_FOUND } from '../../helpers/AppConstants';

const PokemonDetails = ({ data, closeDetails }) => {
  const {
    details: { sprites, id, abilities, height, weight, types },
    name,
  } = data;

  return (
    <div className="overlay">
      <div className="pokemon-details">
        <button onClick={closeDetails} />
        <div className="pokemon-details-inner">
          <aside className="image" style={{ backgroundImage: `url(${sprites.front_default || NOT_FOUND})` }} />
          <aside className="details">
            <h3>#ID: {id}</h3>
            <h4>NAME: {name}</h4>
            <h4>Height: {height} | Weight: {weight}</h4>
            {abilities.length &&
              <section>
                <h4>Abilities</h4>
                {abilities.map(({ ability }, i) => <span key={ability.name + i}>{ability.name}</span>)}
              </section>
            }
            {types.length &&
              <section>
                <h4>Types</h4>
                {types.map(({ type }, i) => <span key={type.name + i}>{type.name}</span>)}
              </section>
            }
          </aside>
        </div>
      </div>
    </div>
  )
};

PokemonDetails.propTypes = {
  data: PropTypes.shape({
    details: PropTypes.shape({
      sprites: PropTypes.shape({
        front_default: PropTypes.string,
      }),
      id: PropTypes.number,
      abilities: PropTypes.arrayOf(PropTypes.any),
      height: PropTypes.number,
      weight: PropTypes.number,
      types: PropTypes.arrayOf(PropTypes.any),
    }),
    name: PropTypes.string,
  }).isRequired,
  closeDetails: PropTypes.func.isRequired,
};


export default PokemonDetails;