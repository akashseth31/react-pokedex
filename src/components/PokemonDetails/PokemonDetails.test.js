import React from 'react';
import { mount } from 'enzyme';
import { data } from '../../helpers/mock-data'
import PokemonDetails from './';
import { NOT_FOUND } from '../../helpers/AppConstants';

describe('PokemonDetails', () => {
  const mockProps = {
    data: data[0],
    closeDetails: jest.fn(),
  }
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<PokemonDetails {...mockProps} />);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correct name", () => {
    expect(wrapper.find('h4').first().text()).toEqual(`NAME: ${mockProps.data.name}`);
  });

  it("should render correct background-image", () => {
    const url = mockProps.data.details.sprites.front_default || NOT_FOUND;
    expect(wrapper.find('aside').first().prop('style')).toHaveProperty('backgroundImage', `url(${url})`);
  });

  it("should call closeDetails", () => {
    wrapper.find('button').first().simulate('click');
    expect(wrapper.props().closeDetails).toHaveBeenCalled();
  });
});