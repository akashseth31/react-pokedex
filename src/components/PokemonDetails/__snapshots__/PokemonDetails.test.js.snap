// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetails renders correctly 1`] = `ReactWrapper {}`;
